src/dotnet/Fable.Compiler/Transforms/Replacements.fs

```diff
@@ -1248,19 +1261,23 @@ let options (com: ICompiler) r (t: Type) (i: CallInfo) (thisArg: Expr option) (a
     //     ccall i "Option" "getValue" [getCallee i; makeBoolConst true] |> Some
     // | "ofObj" | "ofNullable" ->
     //     wrap i.returnType (getCallee i) |> Some
-    // | "isSome" | "isNone" ->
-    //     let op =
-    //         if i.memberName = "isSome"
-    //         then BinaryUnequal
-    //         else BinaryEqual
-    //     let comp =
-    //         ([getCallee i; Fable.Value Fable.Null], op)
-    //         ||> makeEqOp i.range
-    //     match i.returnType with
-    //     | Fable.Boolean _ -> Some comp
-    //     // Hack to fix instance member calls (e.g., myOpt.IsSome)
-    //     // For some reason, F# compiler expects them to be applicable
-    //     | _ -> makeLambda [] comp |> Some
+    | "IsSome", _, [c]
+    | "IsNone", _, [c] ->
+        let op =
+            if i.CompiledName = "IsSome"
+            then BinaryUnequal
+            else BinaryEqual
+
+        makeEqOp r c (Fable.Value (ValueKind.Null (firstGenArg com r i.GenericArgs))) op |> Some
```

Comment:
Now the AST contains `OptionTest`. It's better to use it because it makes it easier to change the option representation if necessary. I'll do that :+1:

Reply:
Oh ok, I didn't know.
Thank you :)